kingpin/actors/packagecloud.py

```diff
@@ -0,0 +1,416 @@
+        deleted_packages = yield self._delete(
+            packages_to_delete=self.option('packages_to_delete'),
+            number_to_keep=self.option('number_to_keep'),
+            repo=self.option('repo'))
+        raise gen.Return(deleted_packages)
+
+
+class DeleteByDate(PackagecloudBase):
+
+    """Deletes packages from a PackageCloud repo older than X.
+
+    Adds additional functionality to the `Delete` class with a `older_than`
+    option.  Only packages older than that number of seconds will be deleted.
+
+    **Options**
+
+    :number_to_keep:
+      Keep at least this number of each package
+      (defaults to *0*)
+
+    :older_than:
+      Delete packages created before this number of seconds
+
+    :packages_to_delete:
+      Regex of packages to delete, e.g. pkg1|pkg2
+
+    :repo:
+      Which packagecloud repo to delete from
+
+    **Examples**
+
+    .. code-block:: json
+
+    { "desc": "packagecloud DeleteByDate example",
+      "actor": "packagecloud.DeleteByDate",
+      "options": {
+          "number_to_keep": 10,
+          "older_than": 600,
+          "packages_to_delete": "deleteme",
+          "repo": "test"
+      }
+    }
+
+    """
+
+    all_options = {
+        'number_to_keep': (
+            int, 0,
+            'Keep at least this number of each package'),
+        'older_than': (
+            int, REQUIRED,
+            'Delete packages created before this number of seconds'),
+        'packages_to_delete': (
+            str, REQUIRED,
+            'Regex of packages to delete, e.g. pkg1|pkg2'),
+        'repo': (str, REQUIRED,
+                 'Which packagecloud repo to delete from')
+    }
+
+    @gen.coroutine
+    def _execute(self):
+        deleted_packages = yield self._delete(
+            packages_to_delete=self.option('packages_to_delete'),
+            number_to_keep=self.option('number_to_keep'),
+            older_than=self.option('older_than'),
+            repo=self.option('repo'))
+        raise gen.Return(deleted_packages)
+
+
+class WaitForPackage(PackagecloudBase):
+
+    """Searches for a package that matches `name` and `version` until found or
+    a timeout occurs.
+
+    **Options**
+
+    :name:
+      Name of the package to search for as a regex
+
+    :version:
+      Version of the package to search for as a regex
+
+    :repo:
+      Which packagecloud repo to delete from
+
+    :sleep:
+      Number of seconds to sleep for between each search
+
+    **Examples**
+
+    .. code-block:: json
+
+    { "desc": "packagecloud WaitForPackage example",
+      "actor": "packagecloud.WaitForPackage",
+      "options": {
+          "name": "findme",
+          "version": "0.1",
+          "repo": "test",
+          "sleep": 10,
+      }
+    }
+
+    """
+
+    all_options = {
+        'name': (
+            str, REQUIRED, 'Name of the package to search for as a regex'),
+        'version': (
+            str, '.*', 'Version of the package to search for as a regex'),
```

Comment:
Is there any valid regex checking here? What happens if the user types in something thats not regex?